File: notification-service/src/main/java/com/leave/management/notification/service/NotificationService.java
package com.leave.management.notification.service;

import com.leave.management.notification.dto.CreateNotificationRequest;
import com.leave.management.notification.dto.NotificationResponse;

import java.util.List;

public interface NotificationService 
{
    NotificationResponse createNotification(CreateNotificationRequest request);

    NotificationResponse getNotificationById(Long id);

    List<NotificationResponse> getNotificationsByEmployeeId(Long employeeId);

    List<NotificationResponse> getUnreadNotificationsByEmployeeId(Long employeeId);

    void markAsRead(Long id);

    void deleteNotification(Long id);
}

File: frontend/src/services/notificationService.js
import axiosInstance from "./api/axiosInstance";
import API_CONFIG from "./api/apiConfig";

const notificationService = {
    createNotification: async (notificationData) => {
        const response = await axiosInstance.post(
            API_CONFIG.endpoints.notifications.base,
            notificationData
        );

        return response.data;
    },

    getNotificationById: async (id) => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.notifications.byId(id)
        );

        return response.data;
    },

    getNotificationsByEmployeeId: async (employeeId) => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.notifications.byEmployee(
                employeeId
            )
        );

        return response.data;
    },

    getUnreadNotificationsByEmployeeId: async (employeeId) => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.notifications.unread(
                employeeId
            )
        );

        return response.data;
    },

    markAsRead: async (id) => {
        const response = await axiosInstance.patch(
            API_CONFIG.endpoints.notifications.markAsRead(id)
        );

        return response.data;
    },

    deleteNotification: async (id) => {
        const response = await axiosInstance.delete(
            API_CONFIG.endpoints.notifications.byId(id)
        );

        return response.data;
    },
};

export default notificationService;